x-pack/plugins/cases/server/client/cases/create.ts

```diff
@@ -56,6 +57,43 @@ export async function throwIfCustomFieldKeysInvalid({
   }
 }
 
+/**
+ * Throws if there are required custom fields missing in the request.
+ */
+export async function throwIfMissingRequiredCustomField({
+  casePostRequest,
+  casesClient,
+}: {
+  casePostRequest: CasePostRequest;
+  casesClient: CasesClient;
+}) {
+  const requestCustomFields = casePostRequest.customFields;
+
+  if (!Array.isArray(requestCustomFields) || !requestCustomFields.length) {
+    return;
+  }
+
+  const configuration = await casesClient.configure.get({ owner: casePostRequest.owner });
```

Comment:
As you said in the description of your PR, I think it is better if you move the call to get the configuration out of the validation functions in this PR.